src/reader/_storage/_base.py

```diff
@@ -52,17 +52,25 @@ class StorageBase:
     chunk_size = 2**8
 
     @wrap_exceptions(message="while opening database")
-    def __init__(self, path: str, timeout: float | None = None):
+    def __init__(self, path: str, read_only: bool, timeout: float | None = None):
         kwargs: dict[str, Any] = {'factory': CONNECTION_CLS}
         if timeout is not None:
             kwargs['timeout'] = timeout
+        if read_only:
+            path = _sqlite_utils.to_sqlite_readonly_uri(path)
+            kwargs['uri'] = True
```

Comment:
How about passing read_only to (and making it an attribute on) LocalConnectionFactory? Then:

* we don't need the read_only property below (Search.__init__ can use it from storage.factory)
* only LocalConnectionFactory needs to know/care about the sqlite3.connect() URI

(effects on make_reader() discussed there)